.github/workflows/InitDev,Unit tests,Integration tests,Package.yml

```diff
@@ -48,14 +49,38 @@ jobs:
       working-directory: ./Ed-Fi-ODS-Implementation/
       shell: pwsh
       run: |
-            $patternName = 'refs/heads/' +  '${{ env.CURRENT_BRANCH }}'
-            Write-Host "Pattern Name is " $patternName -fore GREEN
-            $is_pull_request_branch = 'False'
-            $is_pull_request_branch = git ls-remote --heads origin ${{ env.CURRENT_BRANCH }} | Select-String -Pattern $patternName -SimpleMatch -Quiet
-            if ($is_pull_request_branch -eq $true) {
-              git fetch origin ${{ env.CURRENT_BRANCH }}
-              git checkout ${{ env.CURRENT_BRANCH }}
-            }
+           $odsBranch = '${{ env.REPOSITORY_DISPATCH_BRANCH }}'
+           Write-Host "OdsBranch: $odsBranch"
+           if($odsBranch -ne ''){
+              $patternName = "refs/heads/$odsBranch"
+              $does_corresponding_branch_exist = $false
+              $does_corresponding_branch_exist = git ls-remote --heads origin $odsBranch | Select-String -Pattern $patternName -SimpleMatch -Quiet
+              if ($does_corresponding_branch_exist -eq $true) {
+                Write-Host "Corresponding branch for $odsBranch exists in Implementation repo, so checking it out"
+                git fetch origin $odsBranch
+                git checkout $odsBranch
+              } else {
+                WriteHost "Corresponding branch for $odsBranch does not exist in Implementation repo, so not changing branch checked out"
+              }
+           } else {
+              Write-Host "ref_name: ${{env.REF_NAME}}"
+              $current_branch = ${{env.REF_NAME}}
+              if ($current_branch -like "*/merge"){
```

Comment:
We can use [github.event_name](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) to know if it comes from a PR (github.event_name == 'pull_request'); that way, the intent of this `if` would be much clearer.